examples/objective-c/hello_world/ViewController.m

```diff
@@ -65,38 +65,38 @@ - (void)performRequest {
   self.requestCount++;
   NSLog(@"Starting request to '%@'", _REQUEST_PATH);
 
-  int requestID = self.requestCount;
   // Note: this request will use an http/1.1 stream for the upstream request.
   // The Swift example uses h2. This is done on purpose to test both paths in end-to-end tests
   // in CI.
-  RequestBuilder *builder = [[RequestBuilder alloc] initWithMethod:RequestMethodGet
-                                                            scheme:_REQUEST_SCHEME
-                                                         authority:_REQUEST_AUTHORITY
-                                                              path:_REQUEST_PATH];
-  Request *request = [builder build];
-  ResponseHandler *handler = [[ResponseHandler alloc] initWithQueue:dispatch_get_main_queue()];
+  int requestID = self.requestCount;
+  RequestHeadersBuilder *builder = [[RequestHeadersBuilder alloc] initWithMethod:RequestMethodGet
+                                                                          scheme:_REQUEST_SCHEME
+                                                                       authority:_REQUEST_AUTHORITY
+                                                                            path:_REQUEST_PATH];
+  [builder addUpstreamHttpProtocol:UpstreamHttpProtocolHttp1];
+  RequestHeaders *headers = [builder build];
 
   __weak ViewController *weakSelf = self;
-  [handler onHeaders:^(NSDictionary<NSString *, NSArray<NSString *> *> *headers,
-                       NSInteger statusCode, BOOL endStream) {
-    NSLog(@"Response status (%i): %ld\n%@", requestID, statusCode, headers);
-    NSString *body = [NSString stringWithFormat:@"Status: %ld", statusCode];
+  InactiveStream *inactiveStream = [self.client newStream];
```

Comment:
`StreamPrototype`?

Reply:
No strong opinion, @junr03 do you have thoughts?

Reply:
active/inactive parallels envoy a bit more, but if you prefer Stream and StreamPrototype I don't care too much.

Reply:
I mean Envoy has an `ActiveStream` type. It doesn't have an `InactiveStream` type, nor does it use streams classified as inactive them as a mechanism to produce active streams.

Reply:
I'm open to other possibilities, too, but `InactiveStream` really suggests something other than a suite of callbacks along with a factory mechanism to me. `StreamBase`?

Reply:
`StreamBase` sounds more like a superclass which doesn't really fit here IMO. I'm not opposed to `StreamPrototype`/`Stream` if you and Jose are in agreement

Reply:
lets do StreamPrototype/Stream

Reply:
Done